test/logger_acc.erl

```diff
@@ -0,0 +1,98 @@
+%% @doc A logger report handler that be used to capture expected errors in
+%% tests. The current error report handlers are disabled during the execution
+%% of a function. Afterwards, they are restored and the errors that occered are
+%% returned along with the return value of the fun.
+-module(logger_acc).
+
+%% Public API
+-export([capture/1]).
+
+%% @doc Executes `Fun' and captures all logged errors returns as well as
+%% uncaught errors in `Fun'.
+-spec capture(fun (() -> ResultOfFun)) ->
+    {ok, ResultOfFun, AccumulatedErrors} |
+    {throw | error | exit, Reason, Trace, AccumulatedErrors}
+  when ResultOfFun :: term(),
+       Reason :: term(),
+       Trace :: list(),
+       AccumulatedErrors :: [{error|warning_msg|info_msg, string()} |
+                             {error_report|warning_report|info_report, term()}].
```

Comment:
These need to be updated? From the usage of this, I see it's something like this:

    {error | warning | info, [atom()], string()}

Reply:
I went for `{logger:level(), [atom()], string()|logger:report()}`.